source/CMakeLists.txt

```diff
@@ -77,16 +77,55 @@ else()
   message(STATUS "Will not build AMD GPU support")  
 endif (USE_ROCM_TOOLKIT)
 
+#define USE_ASCEND, find tensorflow abi info
+if (USE_ASCEND)
+  add_definitions("-DHUAWEI_ASCEND")
+  set(DP_VARIANT "ascend")
+endif(USE_ASCEND)
+if (USE_ASCEND AND BUILD_CPP_IF)
+  set(PYTHON_EXECUTABLE "python3.7")
+  set(FIND_TENSORFLOW_INC "import tensorflow as tf; print(tf.sysconfig.get_compile_flags()[0],end='')")
+  execute_process(
+    COMMAND ${PYTHON_EXECUTABLE} "-c" "${FIND_TENSORFLOW_INC}"
+    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+    OUTPUT_VARIABLE OUTPUT_VAR
+    RESULT_VARIABLE RESULT_VAR
+    ERROR_VARIABLE ERROR_VAR
+  )
+  string(REPLACE "-I" "" TensorFlow_INCLUDE_DIRS ${OUTPUT_VAR})
+  if (NOT ${RESULT_VAR} EQUAL 0)
+    message(FATAL_ERROR "Cannot determine tensorflow include dir, error message: ${ERROR_VAR}")
+  endif()
+
+  set(FIND_TENSORFLOW_LINK "import tensorflow as tf; print(tf.sysconfig.get_link_flags()[0],end='')")
+  execute_process(
+    COMMAND ${PYTHON_EXECUTABLE} "-c" "${FIND_TENSORFLOW_LINK}"
+    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+    OUTPUT_VARIABLE OUTPUT_VAR
+    RESULT_VARIABLE RESULT_VAR
+    ERROR_VARIABLE ERROR_VAR
+  )
+  string(REPLACE "-L" "" TensorFlow_LIBRARY_PATH ${OUTPUT_VAR})
+  set (TensorFlowFramework_LIBRARY "${TensorFlow_LIBRARY_PATH}/libtensorflow_framework.so.1")
```

Comment:
Does it mean only TF v1 is supported? Did you try TF v2?

Reply:
At present, many optimizations on Ascend platform are based on TF1.15, some of which are not applicable to TF2.0. We will improve in the future versions.